ambry-router/src/test/java/com.github.ambry.router/MockCryptoService.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.router;
+
+import com.github.ambry.config.CryptoServiceConfig;
+import java.nio.ByteBuffer;
+import java.security.GeneralSecurityException;
+import java.util.concurrent.atomic.AtomicReference;
+import javax.crypto.spec.SecretKeySpec;
+
+
+/**
+ * MockCryptoService to assist in testing exception cases
+ */
+class MockCryptoService extends GCMCryptoService {
```

Comment:
did you try using mockito instead?

Reply:
I don't want to drag this patch anymore. Would like to get it merged asap. Will work on moving it to mockito later. Will create an issue for the same.